### PR TITLE
chore: bump minimum Python version from 2.7 to 3.6

### PR DESCRIPTION
**Pull request by Justintime50**
* Bumps minimum Python version from 2.7 to 3.6
* Bumps all dependencies (removes those that were Python2 only)
* Adds a Makefile for easier environment management
* Simplifies the CI config
* Consolidates dependencies by moving them all to setup.py
* Temporarily comments out the 61 second sleep timer on a shipment test
* Updates README with new Makefile targets

### CODE REVIEW

**Review by nwithan8 — DISMISSED** [review]
Personally, I think we should keep the requirements list in a standalone `.txt`, and simply have `setup.py` read the file.

Regardless, LGTM.

File: README.md [hunk 1]
```diff
@@ -5,13 +5,6 @@
 
 EasyPost is the simple shipping API. You can sign up for an account at <https://easypost.com>.
 
-Requirements
-------------
-
-* [Python](https://www.python.org) 2.7 or 3.3+ (or corresponding [PyPy](https://pypy.org) versions). Note that we only test on Python 2.7 and 3.5+; we strongly recommend against using 3.3.x or 3.4.x as they are no longer supported by many libraries.
```
Comment (jchen293) [inline README.md:11]: Can we keep the Requirements section? So new developers know which Python version their need to use our latest Python CL at least.

Comment (Justintime50) [reply README.md:11]: I see your point, though I'm more of a fan of letting the code do the talking for items like this (requirements can change often), the metadata listed on pypi.org will list the minimum version as will pypi when you try to install it (it'll install a version that works for your env). This is especially important as the README is always the "current" documentation. For instance, people could still use our lib on Python2 but it'll be the last major version, but if we only say on the README it's usable via Python3.6 and newer, people may not try downloading if they're on Python2. With us bumping to 3.6, there's no explicit instructions needed anymore about versions or environments.

Comment (jchen293) [reply README.md:11]: This makes sense, but I think having some requirement detail would be helpful for future customers who want to use our product, so they can see the Python version requirement from the README file directly.

Comment (Justintime50) [reply README.md:11]: Let's tackle this holistically if we are to document each libs version requirements. I have an internal ticket that we can track this work on.

**Review by jchen293 — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

**Review by jchen293 — APPROVED** [review]

**Comment by nwithan8** [comment]
We'll need to remove the 2.7 tests from `required`

**Comment by Justintime50** [comment]
> We'll need to remove the 2.7 tests from `required`

For sure. I plan to ask systems to help with this once @jontsai can take another look here.

**Review by jontsai — APPROVED** [review]

**Pull request merged**